internal/controller/datadogagent/feature/kubernetesstatecore/feature.go

```diff
@@ -121,6 +121,22 @@ func (f *ksmFeature) Configure(dda metav1.Object, ddaSpec *v2alpha1.DatadogAgent
 			}
 			f.customConfigAnnotationValue = hash
 			f.customConfigAnnotationKey = object.GetChecksumAnnotationKey(feature.KubernetesStateCoreIDType)
+		} else {
+			// Generate dynamic checksum for default configuration (based on user provided collectCrMetrics field and whether or not APIServices/CRD metrics are collected)
+			defaultConfigData := map[string]any{
+				"collect_crds":        f.collectCRDMetrics,
+				"collect_apiservices": f.collectAPIServiceMetrics,
+				"collect_cr_metrics":  f.collectCrMetrics,
+			}
+
+			hash, err := comparison.GenerateMD5ForSpec(defaultConfigData)
+			if err != nil {
+				f.logger.Error(err, "couldn't generate hash for default ksm core config")
+			} else {
+				f.logger.V(2).Info("generated default ksm core config hash", "hash", hash, "config", defaultConfigData)
+				f.customConfigAnnotationValue = hash
+				f.customConfigAnnotationKey = object.GetChecksumAnnotationKey(feature.KubernetesStateCoreIDType)
```

Comment:
There’s a small difference between this and what happens when `ddaSpec.Features.KubeStateMetricsCore.Conf != nil`:
* Here, we create the hash annotation only if the `comparison.GenerateMD5ForSpec(…)` call was successful whereas
* if the `if` block, when `comparison.GenerateMD5ForSpec(…)` returns an error, we log the error, but we still generate the hash annotation.

Is it on purpose ?